c_glib/arrow-glib/memory-pool.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <arrow-glib/memory-pool.hpp>
+
+G_BEGIN_DECLS
+
+/**
+ * SECTION: memory-pool
+ * @section_id: memory-pool-classes
+ * @title: memory-pool classes
+ * @include: arrow-glib/arrow-glib.h
+ *
+ * #GArrowMemoryPool is a class for memory allocation.
+ */
+
+typedef struct GArrowMemoryPoolPrivate_ {
+  arrow::MemoryPool* memory_pool;
+} GArrowMemoryPoolPrivate;
+
+enum {
+  PROP_MEMORY_POOL = 1,
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GArrowMemoryPool, garrow_memory_pool, G_TYPE_OBJECT)
+
+#define GARROW_MEMORY_POOL_GET_PRIVATE(obj)        \
+  static_cast<GArrowMemoryPoolPrivate *>(          \
+     garrow_memory_pool_get_instance_private(      \
+       GARROW_MEMORY_POOL(obj)))
+
+static void
+garrow_memory_pool_set_property(GObject *object,
+                                guint prop_id,
+                                const GValue *value,
+                                GParamSpec *pspec)
+{
+  auto priv = GARROW_MEMORY_POOL_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_MEMORY_POOL:
+    priv->memory_pool =
+      static_cast<arrow::MemoryPool *>(g_value_get_pointer(value));
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_memory_pool_get_property(GObject *object,
+                                guint prop_id,
+                                GValue *value,
+                                GParamSpec *pspec)
+{
+  switch (prop_id) {
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_memory_pool_init(GArrowMemoryPool *object)
+{
+  auto priv = GARROW_MEMORY_POOL_GET_PRIVATE(object);
+  new(&priv->memory_pool) arrow::MemoryPool*;
```

Comment:
We don't need this because `priv->memory_pool` is initialized as `NULL` by GLib. 

```suggestion
```